tket/src/Circuit/include/Circuit/ClassicalExpBox.hpp

```diff
@@ -23,13 +23,34 @@
 
 namespace tket {
 
+// Used as an intermediate class in between ClassicalExpBox and Box
+class ClassicalExpBoxPyBase : public Box {
```

Comment:
Since there is no requirement for a `ClassicalExpBox` to be templated on a Python object, I would suggest calling this simply `ClassicalExpBoxBase`. We should however document (for `ClassicalExpBox`) the requirements of the class `T`.